.github/workflows/docker.yaml

```diff
@@ -26,35 +26,39 @@ jobs:
       with:
         submodules: true
 
-    - name: 🧰 Set up Docker Buildx
-      uses: docker/setup-buildx-action@v2
-
     - name: 🐳 Login to GitHub Container Registry
       uses: docker/login-action@v2
       with:
         registry: ghcr.io
         username: ${{ github.actor }}
         password: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: 📝 Prepare docker image metadata
-      id: meta
-      uses: docker/metadata-action@v4
+    - name: ❄ Prepare nix
+      uses: cachix/install-nix-action@v18
       with:
-        images: |
-          ghcr.io/${{ github.repository_owner }}/${{ matrix.target }}
-        tags: |
-          unstable
-          type=semver,pattern={{version}}
+        extra_nix_config: |
+          accept-flake-config = true
 
-    - name: 🔨 Build and maybe push
-      uses: docker/build-push-action@v3
-      with:
-        context: .
+    - name: 🔨 Build images using nix
+      run: |
+        nix build .#docker.${{ matrix.target }} && docker load < ./result
+        echo "FROM ${{matrix.target}}" | docker build \
+          --label org.opencontainers.image.source=https://github.com/input-output-hk/hydra \
+          --label org.opencontainers.image.licenses=Apache-2.0 \
+          --label org.opencontainers.image.created=$(date -Is) \
+          --label org.opencontainers.image.revision=${{github.sha}} \
+          --label org.opencontainers.image.version=unstable \
+          --tag ghcr.io/${{ github.repository_owner }}/${{matrix.target}}:unstable -
+        # Also tag with semver if we are building a tag
+        building_tag=${{github.ref_type == 'tag'}}
+        [[ ${building_tag} = true ]] && \
+          docker tag ghcr.io/input-output-hk/${{matrix.target}}:unstable ghcr.io/input-output-hk/${{matrix.target}}:${{github.ref_name}}
+        docker images
+        docker inspect ghcr.io/input-output-hk/${{matrix.target}}:unstable
+
+    - name: 📤 Maybe push
+      if: ${{ github.ref == 'refs/heads/master' || (github.ref_type == 'tag' && matrix.target != 'hydraw') }}
```

Comment:
maybe 'hydraw' can defined inside an exclusion list?

Reply:
It's the single exclusion in our repo. The target list is (and always will be) very specific to the project. I don't like to over generalize the job (it becomes harder to read & maintain IMO).